client/src/pages/SearchPage.js

```diff
@@ -0,0 +1,129 @@
+import { Link } from 'react-router-dom'
+import styled from 'styled-components'
+import Poster from '../components/Poster'
+import useSearch from '../hooks/useSearch'
+
+export default function SearchPage() {
+  const { query, results, handleSearch } = useSearch()
+
+  return (
+    <Wrapper>
+      <StyledLabel>
+        Suche:
+        <StyledInput
+          type="text"
+          placeholder="Was willst du finden?"
+          onChange={handleSearch}
+          onKeyDown={event =>
+            (event.key === 'Enter' || event.key === 'Escape') &&
+            event.target.blur()
+          }
+        />
+      </StyledLabel>
+      {results !== null && (
+        <>
+          <StyledResultInfo>
+            <strong>{results.length}</strong> Ergebnis
+            {results.length > 1 && `se`} für <i>"{query}"</i> gefunden 🥳
+          </StyledResultInfo>
+          <StyledList>
+            {results.map(
+              ({
+                id,
+                name,
+                poster_path: posterPath,
+                first_air_date: firstAirDate,
+              }) => (
+                <Link to={`/serie/${id}`} key={id}>
+                  <StyledListItem>
+                    <Poster
+                      path={
+                        posterPath
+                          ? `https://image.tmdb.org/t/p/w154${posterPath}`
+                          : 'poster.png'
+                      }
+                      alt={`Poster von ${name}`}
+                    />
+                    <StyledMetaInfos>
+                      <StyledHeading>{name}</StyledHeading>
+                      {firstAirDate !== undefined && (
+                        <div>{firstAirDate.substring(0, 4)}</div>
+                      )}
+                    </StyledMetaInfos>
+                  </StyledListItem>
+                </Link>
+              )
+            )}
+          </StyledList>
+        </>
+      )}
+    </Wrapper>
+  )
+}
+
+const Wrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  padding: 0 16px;
+  gap: 32px;
+`
+
+const StyledLabel = styled.label`
+  display: flex;
+  flex-direction: column;
```

Comment:
Could be archived by simply using `grid`:
```suggestion
  display: grid;
```